src/test/java/org/springframework/integration/kafka/dsl/KafkaDslTests.java

```diff
@@ -178,15 +181,18 @@ public IntegrationFlow sendToKafkaFlow() {
 			return f -> f
 					.<String>split(p -> Stream.generate(() -> p).limit(101).iterator(), null)
 					.publishSubscribeChannel(c -> c
-							.subscribe(sf -> sf.handle(kafkaMessageHandler(producerFactory(), TEST_TOPIC),
-									e -> e.id("kafkaProducer")))
-							.subscribe(sf -> sf.handle(kafkaMessageHandler(producerFactory(), TEST_TOPIC3),
-									e -> e.id("kafkaProducer3")))
+							.subscribe(sf -> sf.handle(
+									kafkaMessageHandler(producerFactory(), TEST_TOPIC1)
+											.timestampExpression("T(Long).valueOf('1487694048633')"),
```

Comment:
FYI. The SpEL is pretty smart with type conversions.
So, there is no reason to explicitly say which type for string you want. The `ConversionService` just easily will use `Long.valueOf()`  for your simple expression like : `.timestampExpression("'1487694048633'")`.
See `StringToNumberConverterFactory`.